tests/system/action/user/test_send_invitation_email.py

```diff
@@ -0,0 +1,524 @@
+from time import time
+
+from openslides_backend.action.mixins.send_email_mixin import (
+    ConnectionSecurity,
+    EmailSettings,
+)
+from tests.system.action.base import BaseActionTestCase
+from tests.system.action.mail_base import AIOHandler, AiosmtpdConnectionManager
+
+
+class SendInvitationMail(BaseActionTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.create_meeting()
+        self.set_models(
+            {
+                "meeting/1": {
+                    "name": "annual general meeting",
+                    "users_email_sender": "Openslides",
+                    "users_pdf_url": "http://example.com:8000",
+                },
+                "user/2": {
+                    "username": "Testuser 2",
+                    "first_name": "Jim",
+                    "last_name": "Beam",
+                    "default_password": "secret",
+                    "email": "recipient2@example.com",
+                    "group_$1_ids": [1],
+                    "meeting_ids": [1],
+                },
+            },
+        )
+        # important to reset all these settings
+        EmailSettings.connection_security = ConnectionSecurity.NONE
+        EmailSettings.host = "127.0.0.1"
+        EmailSettings.port = 25
+        EmailSettings.timeout = 5
+        EmailSettings.user = ""
+        EmailSettings.password = ""
+        EmailSettings.accept_self_signed_certificate = False
+        EmailSettings.default_from_email = "noreply@example.com"
+
+    def test_send_correct(self) -> None:
+        start_time = time()
+        handler = AIOHandler()
+        with AiosmtpdConnectionManager(handler):
```

Comment:
It seems counterintuitive to me to execute the request in a special context for testing purposes, since this does not exist in case of "real" requests. Or how is it handled there?

Reply:
Maybe it's more the name, than the functionality: The "real" request uses a running mail server instance and doesn't need to start and stop it. In case of the test this context manager starts the aiosmtpd test server in a test specific way, i.e. testserver with or without auth (depends the EmailSettings) and especially stops it, depending on it's state.
I'll change the name from AiosmtpdConnectionManager to AiosmtpdServerManager

Reply:
Ah okay, I just did not make the connection. I think both names are fine